src/history/base.js

```diff
@@ -104,7 +110,7 @@ export class History {
       // When the user navigates through history through back/forward buttons
       // we do not want to throw the error. We only throw it if directly calling
       // push/replace. That's why it's not included in isError
-      if (!isExtendedError(NavigationDuplicated, err) && isError(err)) {
+      if (!isRouterError(err, NavigationFailureType.NAVIGATION_DUPLICATED) && isError(err)) {
```

Comment:
We can remove the `isError` check since `isRouterError` already checks that
```suggestion
      if (!isRouterError(err, NavigationFailureType.NAVIGATION_DUPLICATED)) {
```

Reply:
I don't think it's equivalent: If I understood correctly we want to check if `err` is an error but not a navigation duplicated error. When `err` is `null` (like here https://github.com/vuejs/vue-router/blob/588220c475b29e038bc6c0badddac7992bf27445/src/history/base.js#L188) the result of the condition will not be the same.

Reply:
You are right,  misread the line